Tests/YamsTests/EncoderTests.swift

```diff
@@ -0,0 +1,748 @@
+        func testEncodingTopLevelStructuredStruct() {
+            // Address is a struct type with multiple fields.
+            let address = Address.testValue
+            _testRoundTrip(of: address, expectedYAML: """
+                street: 1 Infinite Loop
+                city: Cupertino
+                state: CA
+                zipCode: 95014
+                country: United States
+
+                """)
+        }
+
+        func testEncodingTopLevelStructuredClass() {
+            // Person is a class with multiple fields.
+            let person = Person.testValue
+            _testRoundTrip(of: person, expectedYAML: "name: Johnny Appleseed\nemail: appleseed@apple.com\n")
+        }
+
+        func testEncodingTopLevelStructuredSingleStruct() {
+            // Numbers is a struct which encodes as an array through a single value container.
+            let numbers = Numbers.testValue
+            _testRoundTrip(of: numbers, expectedYAML: "- 4\n- 8\n- 15\n- 16\n- 23\n- 42\n")
+        }
+
+        func testEncodingTopLevelStructuredSingleClass() {
+            // Mapping is a class which encodes as a dictionary through a single value container.
+            let mapping = Mapping.testValue
+            _testRoundTrip(of: mapping, expectedYAML: """
+                Apple:
+                  relative: http://apple.com
+                localhost:
+                  relative: http://127.0.0.1
+
+                """)
+        }
+
+        func testEncodingTopLevelDeepStructuredType() {
+            // Company is a type with fields which are Codable themselves.
+            let company = Company.testValue
+            _testRoundTrip(of: company, expectedYAML: """
+                address:
+                  street: 1 Infinite Loop
+                  city: Cupertino
+                  state: CA
+                  zipCode: 95014
+                  country: United States
+                employees:
+                - id: 42
+                  name: Johnny Appleseed
+                  email: appleseed@apple.com
+
+                """)
+        }
+
+        func testEncodingClassWhichSharesEncoderWithSuper() {
+            // Employee is a type which shares its encoder & decoder with its superclass, Person.
+            let employee = Employee.testValue
+            _testRoundTrip(of: employee, expectedYAML: "id: 42\nname: Johnny Appleseed\nemail: appleseed@apple.com\n")
+        }
+
+        func testEncodingTopLevelNullableType() {
+            // EnhancedBool is a type which encodes either as a Bool or as nil.
+            _testRoundTrip(of: EnhancedBool.true, expectedYAML: "true\n...\n")
+            _testRoundTrip(of: EnhancedBool.false, expectedYAML: "false\n...\n")
+            _testRoundTrip(of: EnhancedBool.fileNotFound, expectedYAML: "null\n...\n")
+        }
+
+        // MARK: - Date Strategy Tests
+        func testEncodingDate() {
+        #if os(Linux)
+            print("'Date' does not conform to 'Codable' on Linux yet.")
```

Comment:
By the way, this is [now fixed](https://github.com/apple/swift-corelibs-foundation/blob/eac219a52fd60a73d50b673b081f88910bd0c339/Foundation/Date.swift#L270) 🎉